File: app/src/main/java/com/krystianwsul/checkme/notifications/InstanceNotificationDeleteService.java
package com.krystianwsul.checkme.notifications;

import android.app.IntentService;
import android.content.Context;
import android.content.Intent;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import com.krystianwsul.checkme.domainmodel.DomainFactory;
import com.krystianwsul.checkme.utils.InstanceKey;
import com.krystianwsul.checkme.utils.TaskKey;

import junit.framework.Assert;

public class InstanceNotificationDeleteService extends IntentService {
    private final static String INSTANCE_KEY = "instanceKey";

    @NonNull
    public static Intent getIntent(@NonNull Context context, @NonNull InstanceKey instanceKey) {
        if (instanceKey.getType() == TaskKey.Type.REMOTE && instanceKey.mScheduleKey.ScheduleTimePair.mCustomTimeKey != null)
            Assert.assertTrue(instanceKey.mScheduleKey.ScheduleTimePair.mCustomTimeKey.getType() == TaskKey.Type.REMOTE);

        Intent intent = new Intent(context, InstanceNotificationDeleteService.class);
        intent.putExtra(INSTANCE_KEY, (Parcelable) instanceKey);
        return intent;
    }

    public InstanceNotificationDeleteService() {
        super("InstanceNotificationDeleteService");
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        InstanceKey instanceKey = intent.getParcelableExtra(INSTANCE_KEY);
        Assert.assertTrue(instanceKey != null);

        DomainFactory.getDomainFactory(this)
                .setInstanceNotified(this, 0, instanceKey);
    }
}

File: app/src/main/java/com/krystianwsul/checkme/notifications/InstanceNotificationDeleteService.kt
package com.krystianwsul.checkme.notifications

import android.app.IntentService
import android.content.Context
import android.content.Intent
import android.os.Parcelable
import com.krystianwsul.checkme.domainmodel.DomainFactory
import com.krystianwsul.checkme.domainmodel.extensions.setInstanceNotified
import com.krystianwsul.checkme.persistencemodel.SaveService
import com.krystianwsul.common.utils.InstanceKey

class InstanceNotificationDeleteService : IntentService("InstanceNotificationDeleteService") {

    companion object {

        private const val INSTANCE_KEY = "instanceKey"

        fun getIntent(context: Context, instanceKey: InstanceKey) = Intent(context, InstanceNotificationDeleteService::class.java).apply {
            putExtra(INSTANCE_KEY, instanceKey as Parcelable)
        }
    }

    override fun onHandleIntent(intent: Intent?) {
        val instanceKey = intent!!.getParcelableExtra<InstanceKey>(INSTANCE_KEY)!!

        DomainFactory.addFirebaseListener {
            it.checkSave()
            it.setInstanceNotified(0, SaveService.Source.SERVICE, instanceKey)
        }
    }
}
